core/arch/arm/kernel/abort.c

```diff
@@ -379,10 +379,6 @@ static void handle_user_mode_panic(struct abort_info *ai)
 	ai->regs->apiakey_hi = tc->keys.apia_hi;
 	ai->regs->apiakey_lo = tc->keys.apia_lo;
 #endif
-
-	daif = (ai->regs->spsr >> SPSR_32_AIF_SHIFT) & SPSR_32_AIF_MASK;
-	/* XXX what about DAIF_D? */
```

Comment:
One thing I am not sure about is that if it is okay to remove this line. It seems like `el0_sync_abort` does not touch the memory at the offset `THREAD_ABG_REG_SPSR`, but I am just wondering if I am missing something. Other than this, it looks good to me and I think it can be squashed into previous commit.

Reply:
`THREAD_ABT_REG_SPSR` is updated at:
https://github.com/OP-TEE/optee_os/blob/f7e4fc1e3c341532465a8e0c602e77d1cc7a3633/core/arch/arm/kernel/thread_a64.S#L1027-L1029

However, there is a problem. We might need to create the new SPSR based on the EL0 SPSR since for instance (some of?) the DAIF bits are masked when taking an exception.

Reply:
I am wondering if we need to keep the previous EL0 SPSR value here. If the `abort_handler()` hits the `handle_user_mode_panic()` then it will never return to the EL0 iiuc, and the previous `spsr` value has been already printed (in `get_fault_type()`).

Reply:
I changed back to the old way of setting the SPSR, except that I preserve the PAN bit in the same way as the DAIF bits.